packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -1318,6 +1324,15 @@ class _RenderDecoration extends RenderBox {
       return box.size.width;
     }
 
+    double customLayout(RenderBox box, double x, double y) {
```

Comment:
I think that this needs to give some indication that it's laying out based on a VerticalAlignment.  Maybe just add a comment above this, like "// Lays out based on y as a VerticalAlignment y value."?  Or rename it to alignedLayout and `y` to `alignmentY` or something?